src/script.lua

```diff
@@ -3089,9 +3124,12 @@ function onCreate(is_new)
 
 		local adminAll = property.checkbox("Admin all players", "false")
 		local everyoneRole = G_roles.get("Everyone")
+
 		if everyoneRole then
 			everyoneRole.setPermissions(adminAll, everyoneRole.auth)
 			G_preferences.adminAll.value = true
+		else
+			companionError("Everyone role not found, can not admin everyone!")
```

Comment:
Hmmm.

Technically, the `everyone` role should **always** exist unless someone deletes it from their savedata files. If this *has* somehow happened, it should probably be a warning in-game.

However, the joining player would not see the message in-game as `onPlayerJoin()` fires before the player's chat has loaded. Maybe it should warn on both. That way if there was no supervisor on the server it can tell the companion but if there is, they will be told in-game.

```suggestion
			companionError("Cannot admin everyone, the Everyone role is missing!")
			tellSupervisors("ERROR", "Cannot admin everyone, the Everyone role is missing!")
```

Reply:
Never mind. This is in `onCreate()` not `onPlayerJoin()`, there is no way anyone would see it, other than the companion 🥴